.github/workflows/documentation_on_main.yml

```diff
@@ -0,0 +1,50 @@
+name: Build dev documentation
+
+on:
+  push:
+    branches:
+      - 'main'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: "Set up Python 3.11"
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
+      - name: "Install pandoc"
+        run: |
+          sudo apt-get install -y pandoc
+      - name: Install sphinx and dependecies
+        run: pip install sphinx==6.2.0 nbsphinx==0.9.1 sphinx-design==0.4.1 pydata-sphinx-theme==0.13.3
+      - name: Install AISee package
+        run: pip install .
+      - name: Build latest documentation
+        run: |
+          cd docs/
+          make copy-examples
+          make html
+      - name: List directories
+        run: ls
+      - name: List branches
+        run: git branch -a
+      - name: Switch to gh-pages branch
+        run: git checkout gh-pages
```

Comment:
When you checkout, I think that you also lose what you generated in `make html` because in the `.gitignore` we are only ignoring `docs/_build/`. We should modify the gitignore if we want this to work.

Reply:
I used this workflow to upload the `dev` version (removing the `on: push: branches: main` restriction) and it works correctly. Even with the `.gitignore`, the `docs/_build/` files are still there. The only problem  would be if you wanted to push the documentation to a non `gh-pages` branch.

Reply:
The generated files are under `docs/build/` or `docs/_build/`?

Reply:
The generated files are under `docs/build/html`